h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -682,6 +667,31 @@ private void setCommonModelBuilderParams(Model.Parameters params) {
     params._keep_cross_validation_fold_assignment = buildSpec.build_control.nfolds != 0 && buildSpec.build_control.keep_cross_validation_fold_assignment;
   }
 
+  private void setStoppingCriteria(Model.Parameters parms, Model.Parameters defaults) {
+    // If the caller hasn't set ModelBuilder stopping criteria, set it from our global criteria.
+
+    //FIXME: Do we really need to compare with defaults before setting the buildSpec value instead?
+    // This can create subtle bugs: e.g. if dev wanted to enforce a stopping criteria for a specific algo/model,
+    // he wouldn't be able to enforce the default value, that would always be overridden by buildSpec.
+    // We should instead provide hooks and ensure that properties are always set in the following order:
+    //  1. defaults, 2. user defined, 3. internal logic/algo specific based on the previous state (esp. handling of AUTO properties).
+    if (parms._stopping_metric == defaults._stopping_metric)
+      parms._stopping_metric = buildSpec.build_control.stopping_criteria.stopping_metric();
+
+    if (parms._stopping_metric == StoppingMetric.AUTO) {
+      String sort_metric = getSortMetric();
+      parms._stopping_metric = sort_metric == null ? StoppingMetric.AUTO
+                              : sort_metric.equals("auc") ? StoppingMetric.logloss
+                              : metricValueOf(sort_metric);
```

Comment:
@ledell , please confirm this is the expected behaviour:
1. if stopping metrics specified by user (different from AUTO), then use it everywhere (models, grids).
2. otherwise, if no sort_metric set by user, use AUTO stopping metric first.
   leaderboard with update the sort_metric once the first model has been added, so it won't be null anymore.
3. if sort_metric set to "auc" by user or by leaderboard, then set stopping metric to logloss.
3. for other sort metrics, use the matching stopping metric